sceptre_cdk_handler/cdk_builder.py

```diff
@@ -133,33 +129,11 @@ def _get_envs(self) -> Dict[str, str]:
         Returns:
             The dictionary of environment variables.
         """
-        envs = self._environment_variables.copy()
-        envs.pop("AWS_PROFILE", None)
-        # Set aws environment variables specific to whatever AWS configuration has been set on the
-        # stack's connection manager.
-        credentials: Credentials = self._connection_manager._get_session(
-            self._connection_manager.profile,
-            self._connection_manager.region,
-            self._connection_manager.iam_role
-        ).get_credentials()
+        envs = self._connection_manager.create_session_environment_variables()
         envs.update(
-            AWS_ACCESS_KEY_ID=credentials.access_key,
-            AWS_SECRET_ACCESS_KEY=credentials.secret_key,
-            # Most AWS SDKs use AWS_DEFAULT_REGION for the region
-            AWS_DEFAULT_REGION=self._connection_manager.region,
             # CDK frequently uses CDK_DEFAULT_REGION in its docs
             CDK_DEFAULT_REGION=self._connection_manager.region,
```

Comment:
should the handler also set the `CDK_DEFAULT_ACCOUNT` variable?  https://docs.aws.amazon.com/cdk/v2/guide/environments.html

Reply:
So Sceptre never has a default awareness of an "account". You can pass that data in via a `--var`, but it never deals with accounts otherwise. So, my thoughts to this are:
* If you want to pass an account into CDK stack, you can do that either via the context or via the sceptre_user_data
* If you're using the bootstrap stack, I _believe_ that the bootstrap stack will automatically imply the correct account.

Anyway, we could hit STS's get-caller-identity and get the account id to pass in... but I don't really see the benefit. Do you?

Reply:
it's not a big deal.  the work arounds sounds fine to me.